app/src/main/java/de/xikolo/viewmodels/login/LoginViewModel.kt

```diff
@@ -0,0 +1,31 @@
+package de.xikolo.viewmodels.login
+
+import de.xikolo.network.jobs.CreateAccessTokenJob
+import de.xikolo.network.jobs.base.NetworkStateLiveData
+import de.xikolo.viewmodels.base.BaseViewModel
+import de.xikolo.viewmodels.shared.UserDelegate
+
+class LoginViewModel : BaseViewModel() {
+
+    private val userDelegate = UserDelegate(realm)
+
+    val loginNetworkState: NetworkStateLiveData = networkState
+    val profileNetworkState: NetworkStateLiveData by lazy {
+        NetworkStateLiveData()
+    }
```

Comment:
I think in this case the network state object can be shared by the two sequential requests. Currently, the `profileNetworkState` is useless since it is not connected to the UI. Passing null would do the same.

Reply:
Solved